src/briefcase/platforms/linux/system.py

```diff
@@ -163,6 +163,21 @@ def finalize_app_config(self, app: AppConfig):
 
         if not self.use_docker:
             app.target_image = f"{app.target_vendor}:{app.target_codename}"
+        else:
+            if app.target_vendor_base == ARCH and self.tools.docker.is_users_mapped:
+                raise BriefcaseCommandError(
```

Comment:
This'll provide feedback when Arch package builds aren't feasible....but it also better illuminates that macOS can no longer build Arch packages...

Reply:
Well that's annoying... if I'm reading this right, this branch will also catch *Linux* users whose Docker config doesn't require user mapping. 

Would a better approach be to *always* turn on the step-down user for arch builds, and only raise an error if the local configuration will *break* with a step down user?

(Mostly asking for self-serving purposes - I can't easily build Arch packages without this capability)

Reply:
(chatted a lot off line but answering here for posterity)

> Well that's annoying... if I'm reading this right, this branch will also catch _Linux_ users whose Docker config doesn't require user mapping.

This will catch Linux users using Docker Desktop or rootless Docker since they can't use the step-down user and would be running as `root` in the container now. Existing users with Docker Engine wouldn't be affected since user mapping isn't (normally) enabled.
 
> Would a better approach be to _always_ turn on the step-down user for arch builds, and only raise an error if the local configuration will _break_ with a step down user?
> 
> (Mostly asking for self-serving purposes - I can't easily build Arch packages without this capability)

We can only use the step-down user in containers when Docker is _not_ performing user mapping. This is because of how user mapping works. If Docker is performing user mapping and we use a user in the container with ID 1000, then the owner of the files in the host file system end up with an owner ID that doesn't even exist; however, this user ID that shows up is tied to how Linux user namespaces work.

Nonetheless, this also reveals another strategy that could allow us to accommodate this. We could set the GID of all the files to a GID the host user is also a part of. In that way, while the files are owned by a "non-existent" user, the host user would still have permission to interact with them. I ultimately didn't pursue this, though, because it presents a lot of opportunity for issues one way or another. In general, a lot of the strategies to mitigate these file permissions problems required strong cooperation between the host environment and containers that I thought would be onerous for users.